contracts/extensions/VotingReputation.sol

```diff
@@ -0,0 +1,768 @@
+
+  // Public functions (interface)
+
+  function createRootPoll(
+    address _target,
+    bytes memory _action,
+    bytes memory _key,
+    bytes memory _value,
+    uint256 _branchMask,
+    bytes32[] memory _siblings
+  )
+    public
+  {
+    require(state == ExtensionState.Active, "voting-rep-not-active");
+
+    uint256 rootSkillId = colony.getDomain(1).skillId;
+    createPoll(_target, _action, rootSkillId, _key, _value, _branchMask, _siblings);
+  }
+
+  function createDomainPoll(
+    uint256 _domainId,
+    uint256 _childSkillIndex,
+    address _target,
+    bytes memory _action,
+    bytes memory _key,
+    bytes memory _value,
+    uint256 _branchMask,
+    bytes32[] memory _siblings
+  )
+    public
+  {
+    require(state == ExtensionState.Active, "voting-rep-not-active");
+
+    uint256 domainSkillId = colony.getDomain(_domainId).skillId;
+    uint256 actionDomainSkillId = getActionDomainSkillId(_action);
+
+    if (domainSkillId != actionDomainSkillId) {
+      uint256 childSkillId = colonyNetwork.getChildSkillId(domainSkillId, _childSkillIndex);
+      require(childSkillId == actionDomainSkillId, "voting-rep-invalid-domain-id");
+    }
+
+    createPoll(_target, _action, domainSkillId, _key, _value, _branchMask, _siblings);
+  }
+
+  function stakePoll(
+    uint256 _pollId,
+    uint256 _permissionDomainId, // For extension's arbitration permission
+    uint256 _childSkillIndex, // For extension's arbitration permission
+    uint256 _domainId,
+    bool _vote,
+    uint256 _amount,
+    bytes memory _key,
+    bytes memory _value,
+    uint256 _branchMask,
+    bytes32[] memory _siblings
+  )
+    public
+  {
+    Poll storage poll = polls[_pollId];
+
+    require(
+      activePolls[hashAction(poll.action)] == 0 ||
+      activePolls[hashAction(poll.action)] == _pollId,
+      "voting-rep-competing-poll-exists"
+    );
+
+    uint256 stakerRep = checkReputation(_pollId, msg.sender, _key, _value, _branchMask, _siblings);
+
+    // TODO: can we keep the domainId on the poll somewhere? This seems like a wasteful external call.
+    //   But if it's 10 external calls per word of storage, then < 10 stakers makes this cheaper.
+    require(colony.getDomain(_domainId).skillId == poll.skillId, "voting-rep-bad-domain-id");
+    require(add(stakes[_pollId][msg.sender][_vote], _amount) <= stakerRep, "voting-rep-insufficient-rep");
+
+    require(getPollState(_pollId) == PollState.Staking, "voting-rep-staking-closed");
+    require(add(poll.stakes[toInt(_vote)], _amount) <= getRequiredStake(_pollId), "voting-rep-stake-too-large");
+
+    require(
+      crowdFunding || add(poll.stakes[toInt(_vote)], _amount) == getRequiredStake(_pollId),
+      "voting-rep-stake-must-be-exact"
+    );
+
+    colony.obligateStake(msg.sender, _domainId, _amount);
+    colony.transferStake(_permissionDomainId, _childSkillIndex, address(this), msg.sender, _domainId, _amount, address(this));
```

Comment:
It's a good point... it's something I've thought about. I think the most likely alternative would involve the type of "matching table" you describe in the whitepaper appendix for matching stakes for and against. If we had a table like that, we could leave the stake undisturbed until after the vote. Unfortunately it would increase the gas cost of staking. Also, it likely wouldn't extend to the voter compensation, which would almost certainly need to be taken out at the time of stake (although being able to hold on to 90% of your stake is way better than 0%).

Reply:
Will discuss with research

Reply:
Ok, so there's another hiccup here. Since staking is managed on the per-domain level (for security reasons), and since poll escalation means that stakes may be made across multiple domains, letting stakers retain their stakes means that all rewards claims will require reconciling the stakes for every level of escalation separately. This adds a new level of complexity and bookkeeping which will need to be worked through. Given our timeline I would strongly recommend we punt this until a future version of the voting extension and release it as-is, even though the mechanics now create a known disincentive to stake.